Dockerfile.cugraph

```diff
@@ -0,0 +1,120 @@
+    build-essential `mage-memgraph` \
+    cmake           `mage-memgraph` \
+    curl            `mage-memgraph` \
+    g++             `mage-memgraph` \
+    python3         `mage-memgraph` \
+    python3-pip     `mage-memgraph` \
+    python3-setuptools     `mage-memgraph` \
+    python3-dev     `mage-memgraph` \
+    uuid-dev        `mage-memgraph` \
+    clang           `mage-memgraph` \
+    git             `mage-memgraph` \
+    --no-install-recommends \
+    # Download and install Memgraph
+    && curl https://download.memgraph.com/memgraph/v${MG_VERSION}/ubuntu-20.04/memgraph_${MG_VERSION}-1_amd64.deb --output memgraph.deb \
+    && dpkg -i memgraph.deb \
+    && rm memgraph.deb \
+    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+# Memgraph listens for Bolt Protocol on this port by default.
+EXPOSE 7687
+
+FROM cugraph-dev as dev
+
+# Copy and build MAGE
+WORKDIR /mage
+COPY . /mage
+
+RUN curl https://sh.rustup.rs -sSf | sh -s -- -y && \
+    export PATH="/root/.cargo/bin:${PATH}" && \ 
+    python3 -m  pip install -r /mage/python/requirements.txt && \  
+    python3 /mage/setup build \ 
+            --gpu \
+            --cpp-build-flags MAGE_CUGRAPH_ROOT=/opt/conda/envs/rapids/ \
+            -p /usr/lib/memgraph/query_modules/ 
+
+USER memgraph
+ENTRYPOINT ["/usr/lib/memgraph/memgraph"]
+
+FROM nvidia/cuda:${CUDA_VERSION}.0-runtime-ubuntu20.04 AS prod
+
+USER root
+ARG MG_VERSION
+ARG PY_VERSION
+ENV MG_VERSION ${MG_VERSION}
+ENV PY_VERSION ${PY_VERSION}
+
+# Copy modules
+COPY --from=dev /usr/lib/memgraph/query_modules/ /usr/lib/memgraph/query_modules/
+# Copy cugraph library
+COPY --from=dev /opt/conda/envs/rapids/lib/libcugraph.so /opt/conda/envs/rapids/lib/libcugraph.so
+# Copy python build
+COPY --from=dev /usr/local/lib/python${PY_VERSION}/ /usr/local/lib/python${PY_VERSION}/
+
+# NVIDIA key rotation
+RUN rm /etc/apt/sources.list.d/cuda.list
+
+# Download and install Memgraph
+RUN apt-get update && apt-get install -y \
+    libcurl4        `memgraph` \
+    libpython${PY_VERSION}   `memgraph` \
+    libssl1.1       `memgraph` \
+    libssl-dev       `memgraph` \
+    openssl         `memgraph` \
+    curl            `mage-memgraph` \
+    libgomp1         `mage-memgraph` \
+    python3         `mage-memgraph` \
+    python3-pip         `mage-memgraph` \
+    python3-setuptools     `mage-memgraph` \
+    python3-dev     `mage-memgraph` \
+    && curl https://download.memgraph.com/memgraph/v${MG_VERSION}/ubuntu-20.04/memgraph_${MG_VERSION}-1_amd64.deb --output memgraph.deb \
+    && dpkg -i memgraph.deb \
+    && rm memgraph.deb \
+    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+RUN export PATH="/usr/local/lib/python${PY_VERSION}:${PATH}"
+
+RUN rm -rf /mage \
+    && export PATH="/usr/local/lib/python${PY_VERSION}:${PATH}" \
+    && apt-get -y --purge autoremove git curl python3-pip python3-dev cmake build-essential \
+    && apt-get clean
+
+
+USER memgraph
+ENTRYPOINT ["/usr/lib/memgraph/memgraph"]
```

Comment:
Let's restructure `Dockerfile.cugraph` image by removing 3 phases:
1. I don't think we need 3 phases here because we have a different situation from one in standard `Dockerfile`. We can have one phase where we install C++, Python, Rust and cuGraph files (that we can do in `rapidsai/rapidsai-dev` image). And the second phase will then be `nvidia/cuda`, where we will copy files we installed in our last phase to this one. I think you should notice that we don't need 3 phases as in a regular `Dockerfile` because our `production` and `dev` image will not share the same base image (`dev` will use `rapidsai/rapidsai-dev`, and `prod` will use `nvidia/cuda`).

Reply:
Completely agreed, that's what I indented to suggest. Furthermore, I would remove dev image for this purpose because we do not need to allow for changing cuGraph algorithms currently.